catalog/core/catalog-core-ftp/src/main/java/ddf/catalog/core/ftp/ftplets/CustomFtplet.java

```diff
@@ -0,0 +1,368 @@
+            String fileName = request.getArgument();
+
+            try {
+                ftpFile = session.getFileSystemView()
+                        .getFile(fileName);
+            } catch (FtpException e) {
+                LOGGER.error("Failed to retrieve file from FTP session");
+            }
+
+            if (ftpFile == null) {
+                LOGGER.error(
+                        "Sending FTP status code 501 to client - syntax errors in request parameters");
+                session.write(new DefaultFtpReply(FtpReply.REPLY_501_SYNTAX_ERROR_IN_PARAMETERS_OR_ARGUMENTS,
+                        "STOR"));
+
+                throw new NullPointerException("File to be transferred from client did not exist");
+            }
+
+            DataConnectionFactory connFactory = session.getDataConnection();
+            if (connFactory instanceof IODataConnectionFactory) {
+                InetAddress address = ((IODataConnectionFactory) connFactory).getInetAddress();
+                if (address == null) {
+                    session.write(new DefaultFtpReply(FtpReply.REPLY_503_BAD_SEQUENCE_OF_COMMANDS,
+                            "PORT or PASV must be issued first"));
+                    LOGGER.error(
+                            "Sending FTP status code 503 to client - PORT or PASV must be issued before STOR");
+                    throw new NullPointerException("FTP client address was null");
+                }
+            }
+
+            if (!ftpFile.isWritable()) {
+                session.write(new DefaultFtpReply(FtpReply.REPLY_550_REQUESTED_ACTION_NOT_TAKEN,
+                        "Insufficient permissions"));
+                LOGGER.error(
+                        "Sending FTP status code 550 to client - insufficient permissions to write file.");
+                throw new FtpException("Insufficient permissions to write file");
+            }
+
+            session.write(new DefaultFtpReply(FtpReply.REPLY_150_FILE_STATUS_OKAY,
+                    "STOR " + fileName));
+            LOGGER.debug("Replying to client with code 150 - file status okay");
+
+            DataConnection dataConnection;
+            FileBackedOutputStream outputStream = new FileBackedOutputStream(1000000);
+            try {
+                dataConnection = connFactory.openConnection();
+                dataConnection.transferFromClient(session, outputStream);
+                session.write(new DefaultFtpReply(FtpReply.REPLY_226_CLOSING_DATA_CONNECTION,
+                        "Closing data connection"));
+                LOGGER.debug("Sending FTP status code 226 to client - closing data connection");
+
+            } catch (Exception e) {
+                LOGGER.error("Error getting the ouput data stream from the FTP session");
+                throw new IOException("Error getting the output stream from FTP session");
+            } finally {
+                session.getDataConnection()
+                        .closeDataConnection();
```

Comment:
Is this a different DataConnection than the one set to the dataConnection variable? If so, does "dataConnection" need closed as well?
Also, might want to use the try with resources syntax so the outputStream is automatically closed:
try (FileBackedOutputStream outputStream = new FileBackedOutputStream(1000000)) { 
...